tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -3065,4 +3065,171 @@ protected function assertEqualMarkup( $expected, $actual, $message = '' ) {
 	protected function add_html5_script_theme_support() {
 		add_theme_support( 'html5', array( 'script' ) );
 	}
+
+	/**
+	 * Test that a script is moved to the footer if it is made non deferrable, was in the header and
+	 * all scripts that depend on it are in the footer.
+	 *
+	 * @ticket 58599
+	 *
+	 * @dataProvider data_provider_script_move_to_footer
+	 *
+	 * @param string $b_in_footer in_footer value for script B.
+	 * @param string $expected    Expected output.
+	 * @param bool   $expect_a_in_footer Is script A in the footer?
+	 * @param bool   $expect_b_in_footer Is script B in the footer?
```

Comment:
Looks like the args need to be updated. The `$expected` arg is missing, as is `$c_in_footer` and `$d_in_footer`.

Reply:
I'm going to try re-arranging this a bit. Working on this now.

Reply:
ok thanks, got a bit messy as it grew :)